src/middlewared/middlewared/plugins/zfs_/pool_actions.py

```diff
@@ -238,6 +239,43 @@ def import_pool(self, name_or_guid, properties, any_host, cachefile, new_name, i
                         'Failed to mount datasets after importing "%s" pool: %s', name_or_guid, str(e), exc_info=True
                     )
 
+    @accepts(Str('pool_name'))
+    def ddt_prefetch(self, pool_name):
+        with libzfs.ZFS() as zfs:
+            try:
+                zfs.get(pool_name).ddt_prefetch()
+            except libzfs.ZFSException as e:
+                raise CallError(str(e), e.code)
+
+    def ddt_prefetch_pools(self):
+        # We have this method so to avoid making excessive calls to process pool service
+        for pool_info in (self.middleware.call('zfs.pool.query_imported_fast')).values():
+            try:
+                self.middleware.logger.info('Prefetching ddt table of %r pool', pool_info['name'])
+                self.ddt_prefetch(pool_info['name'])
+            except CallError as e:
+                self.logger.error('Failed to prefetch DDT for pool %r: %s', pool_info['name'], e)
+
+    @accepts(
```

Comment:
This should use new API and I don't want more than 1 positional arg. Please make it accept a dictionary only